pkg/distributor/distributor.go

```diff
@@ -858,6 +859,100 @@ func (d *Distributor) PushWithCleanup(ctx context.Context, req *mimirpb.WriteReq
 	return &mimirpb.WriteResponse{}, firstPartialErr
 }
 
+// StartBackfill requests the starting of a backfill session.
+func (d *Distributor) StartBackfill(ctx context.Context, tenantID int, blockID string) error {
+	level.Info(d.log).Log("msg", "starting backfill", "tenantId", tenantID, "blockId", blockID)
+	// TODO: Verify that block hasn't already been ingested
+	return nil
+}
+
+// AddBackfillFile adds a file to an ongoing metrics backfill.
+func (d *Distributor) AddBackfillFile(ctx context.Context, tenantID int, blockID, pth string, r *http.Request) error {
+	tenantIDFromCtx, err := tenant.TenantID(ctx)
+	if err != nil {
+		return err
+	}
+	localCtx := user.InjectOrgID(ctx, tenantIDFromCtx)
+	// Get clientIP(s) from Context and add it to localCtx
+	source := util.GetSourceIPsFromOutgoingCtx(ctx)
```

Comment:
I see quite a lot of complexity related to the propagation of the source IP. I don't think we need all this logic at all. If I remember correctly, Mimir doesn't even use it. I would remove all of it, and reconsider in a second step.

Reply:
Thanks, that's OK by me. I just copied from existing code without knowing if it's useful or not, was meaning to return to it in an iteration.